src/mergeDiff.js

```diff
@@ -40,7 +41,9 @@ function mergeDiffArr(arrA, arrB, collB, indexA, indexB, onRemove, accum) {
   return mergeDiffArr(arrA, arrB, collB, indexA + 1, indexB, onRemove, accum);
 }
 
-export default function mergeDiff(a, b, onRemove) {
+export default function mergeDiff(a: Object,
+                                  b: Object,
+                                  onRemove: (key: string) => Object): Object {
```

Comment:
`mergeDiff` is called only once in `components.js`. It could be possible to inline `onRemove`.